### DISCUSSION TITLE
Infinity loop is happen to tried to setValue() or update() of useFieldArray from same field specific key

### DISCUSSION THREAD

**Discussion by tomoyukikino** [react-hook-form/react-hook-form#9518] [Q&A] [answered]
### Version Number
7.40.0

### Codesandbox/Expo snack
https://codesandbox.io/s/usefieldarray-react-bootstrap-forked-4czkk5?file=/src/CartsFieldArrayForm.tsx

### Steps to reproduce
1. [Go to CartsFieldArrayForm.tsx in codesandbox](https://codesandbox.io/s/usefieldarray-react-bootstrap-forked-4czkk5?file=/src/CartsFieldArrayForm.tsx)
2. uncomment setValue line 46
3. infinity loop happen!!

<img width="2557" alt="Screenshot 2022-12-05 at 9 12 09" src="https://user-images.githubusercontent.com/45547251/205524984-4cf24ebf-be02-4687-9728-1d78519bf029.png">

https://user-images.githubusercontent.com/45547251/205525046-afdc813e-4669-46c9-94ed-2a09cc061248.mov

### My Question

Infinity loop is happen to tried to setValue() or update() of useFieldArray from same field specific key.

Should I give up to used update() or setValue() if I want update same field another key from same field specific key?

This problem can solve to use React.useState.
But I thought I should to use setValue() or update() provided from react-hook-form to more good code.

Perhaps, react-hook-form spec don't allow update data acquired from useWatch to another key at same field used update() of useFieldArray or setValue().

Another option: This problem can solve 'carts.[].price' move out 'carts' field and rename 'cartsPrice.[].price' to prevent update same field key.

**Comment by bluebill1049** [comment, marked as answer]
Have you seen this example: https://codesandbox.io/s/usefieldarray-llp6lw? I think a better approach than set input value at `useEffect`, in fact i would almost try to avoid update value from `useEffect` instead of event handler.

**Reply by tomoyukikino** [reply]
@bluebill1049 
thank you reply me. I try but failed.
That’s what I thought,Cannot use setValue?

https://codesandbox.io/s/usefieldarray-forked-k66rou?file=/src/App.tsx

**Reply by bluebill1049** [reply]
q: when do you want to set the price? 

on form mount or base on user's certain action?

**Reply by tomoyukikino** [reply]
@bluebill1049 
Ideally, user inputted the form. So, onChange the quantity or unitPrice form in useFieldArray.

**Reply by bluebill1049** [reply]
so why not `setValue` on `onChange` on the input itself?

**Reply by tomoyukikino** [reply]
Im fool. That right. I try implement now.

**Reply by bluebill1049** [reply]
👍 no worries. `useEffect` is quite powerful.

> great power comes with great responsibility

so next time seek for event handler first before approach `useEffect`

**Reply by tomoyukikino** [reply]
### Solve with `useState`
thank you @bluebill1049 
I can solved this problem for used `useState` not `useEffect`.
I inputted to quantity or unitPrice form, as a result, displayed calculated price in price form.
codesandbox: https://codesandbox.io/s/usefieldarray-forked-k66rou?file=/src/App.tsx

https://user-images.githubusercontent.com/45547251/205808851-7d27d0b6-f5d2-4700-883c-81cdf9f85021.mov

<img width="483" alt="Screenshot 2022-12-06 at 12 31 01" src="https://user-images.githubusercontent.com/45547251/205806247-b351f93f-f71c-4dd0-a1b8-9894ac611cc1.png">

<img width="547" alt="Screenshot 2022-12-06 at 12 31 05" src="https://user-images.githubusercontent.com/45547251/205806239-d25cae84-e507-4fce-896e-a19af1bd113f.png">

### Quention
I didn't quite understand how to solved with `useEffect`😂
I tried implement, but cannot work.
codesandbox: https://codesandbox.io/s/usefieldarray-useeffect-qwtijk?file=/src/App.tsx

<img width="638" alt="Screenshot 2022-12-06 at 13 01 32" src="https://user-images.githubusercontent.com/45547251/205810105-8887afb4-05d2-4588-b4b9-52cad85ed62c.png">

### Note
I tried to use `getValues`, but not good work.
But I recognize that use `getValues` in this place may mistake.
<img width="588" alt="Screenshot 2022-12-06 at 12 58 40" src="https://user-images.githubusercontent.com/45547251/205811213-98974c1e-71a4-4b22-a386-be39e7ea5c76.png">
